### DISCUSSION TITLE
How to reduce unuse CSS?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#65169] [Help]
### Summary

So I got this error in the Google lighthouse

![image](https://github.com/vercel/next.js/assets/93617433/d3a1a39e-822f-4904-9d94-29c122ea7e75)

And then followed instructions... but still giving me the error after doing another npm build and npm start..

created a postcss.config.js at the root

```
module.exports = {
  plugins: [
    "postcss-flexbugs-fixes",
    [
      "postcss-preset-env",
      {
        autoprefixer: {
          flexbox: "no-2009",
        },
        stage: 3,
        features: {
          "custom-properties": false,
        },
      },
    ],
    [
      "@fullhuman/postcss-purgecss",
      {
        content: [
          "./pages/**/*.{js,jsx,ts,tsx}",
          "./components/**/*.{js,jsx,ts,tsx}",
        ],
        defaultExtractor: (content) => content.match(/[\w-/:]+(?<!:)/g) || [],
        safelist: ["html", "body"],
      },
    ],
  ],
};

```
and installed this:

`npm install postcss-flexbugs-fixes postcss-preset-env.`

^^^  Is there anything else I need to do to get rid of this error?

**Comment by Elindo586** [comment]
BTW...  I haven't open another post about these other ones, but they seem to go together:

minify JavaScript 
Reduce unuse JavaScript  <-  for this one I am using Bundle Analyzer but it isn't working either.

^^^  These 2 with the unuse CCS  seem to be taking a performance value from the mobile page...  the errors show on the web pages from Google lighthouse, but doesn't seem to be taking much performance out of the app.

**Comment by icyJoseph** [comment]
Hi,

For this you can use the coverage tool. There's typically, not a magic command that just makes these go away, unless you have a framework that's built to drop unused CSS. PostCSS will also only go through the files you specified.

You can start by finding the coverage report tool in Chrome Dev Tool, and reloading the page, it'll show you what JavaScript and CSS it not being used. 

<img width="939" alt="Screenshot 2024-04-29 at 22 31 46" src="https://github.com/vercel/next.js/assets/21013447/155eee35-4101-4a8a-8407-7aa62d296831">

Part of frontend development is understanding what CSS needs to be dropped, and having JavaScript budgets. 

To get a real sense of the app performance toward users, https://pagespeed.web.dev/ is  good resource.

**Reply by icyJoseph** [reply]
That being said, I see you import the entire bootstrap CSS into your project. That's bound to contain way more CSS than you ever need. 

Regarding JavaScript, it depends, it might be that you are accidentally importing a package that you only want in the server, into the client... you have to use the coverage tool and the bundle analyzer to figure out what is it that you are sending to the client.
